pkg/handlers/index/alias_es.go

```diff
@@ -0,0 +1,191 @@
+package index
+
+import (
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+	"github.com/zinclabs/zinc/pkg/core"
+	"github.com/zinclabs/zinc/pkg/meta"
+	"github.com/zinclabs/zinc/pkg/zutils"
+)
+
+type Alias struct {
+	Actions []Action `json:"actions"`
+}
+
+type Action struct {
+	Add    *base `json:"add"`
+	Remove *base `json:"remove"`
+}
+
+type base struct {
+	Index   string   `json:"index"`
+	Alias   string   `json:"alias"`
+	Indices []string `json:"indices"`
+	Aliases []string `json:"aliases"`
+}
+
+func AddOrRemoveESAlias(c *gin.Context) {
+	var alias Alias
+	err := zutils.GinBindJSON(c, &alias)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, meta.HTTPResponseError{Error: err.Error()})
+		return
+	}
+
+	addMap := map[string][]string{}
+	removeMap := map[string][]string{}
+
+	indexList := core.ZINC_INDEX_LIST.List()
+
+outerLoop:
+	for _, action := range alias.Actions {
+		if action.Add != nil {
+			if action.Add.Index != "" {
+				matchAndAddToMap(indexList, action.Add.Index, addMap, action.Add)
+				continue outerLoop
+			}
+
+			// index is empty, try the indices field
+			for _, indexName := range action.Add.Indices {
+				matchAndAddToMap(indexList, indexName, addMap, action.Add)
+			}
+
+			continue outerLoop // this was an add action, don't bother checking action.Remove
```

Comment:
both add and remove actions can be in the same request.

Reply:
Actually, In a list of actions, each action can only be either add or remove, not both.